src/main/scala/optics/poly/functions/Index.scala

```diff
@@ -0,0 +1,68 @@
+package optics.poly.functions
+import optics.poly.{Iso, Lens, EOptional}
+
+import scala.annotation.implicitNotFound
+import scala.collection.immutable.{ListMap, SortedMap, Map}
+
+trait Index[Error, From, -Key, To] {
```

Comment:
I wonder about the `Error` being a type param. Say you have 2 instances with the same `From`, `Key` and `To` but different `Errors`. How would you be able to choose when you call `Index.apply`